Assets/ConsentManagementProvider/Scripts/json/JsonUnwrapper.cs

```diff
@@ -32,14 +32,17 @@ public static SpConsents UnwrapSpConsentsAndroid(string json)
             }
             catch (Exception ex)
             {
-                throw new ApplicationException("An error occurred during JSON unwrapping.", ex);
+                throw new ApplicationException("An error occurred during JSON unwrapping." + ex.Message, ex);
             }
         }
 
         private static SpCcpaConsent UnwrapSpCcpaConsentAndroid(CcpaConsentWrapper wrappedCcpa)
         {
             if (wrappedCcpa == null)
-                throw new ArgumentNullException(nameof(wrappedCcpa), "The CCPA consent wrapper cannot be null.");
+            {
+                CmpDebugUtil.LogError("The CCPA consent wrapper cannot be null.");
+                return null;
```

Comment:
why do we want to return null instead of throwing an error? I mean, what do we do with the null value after that is being returned from this function? do we do something to handle this case?

Reply:
This method is a part of unparsing process and I don't mind having the object as null since it is a possible outcome for that object with doesn't break the overall flow. Error handling exists above level in the call tree. The logging is enough for now. Since throwing error triggers entire unparsing process to throw.